### PR TITLE
ddl: add distributed framework

### PR DESCRIPTION
**Pull request by ran-huang**
### What is changed, added or deleted? (Required)

add introduction doc for backend task distributed execution framework document.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13705, https://github.com/pingcap/docs-cn/pull/14001
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -956,6 +956,8 @@
   - [Errors Codes](/error-codes.md)
   - [Table Filter](/table-filter.md)
   - [Schedule Replicas by Topology Labels](/schedule-replicas-by-topology-labels.md)
+  - Internal components
```
Comment (qiancai) [inline TOC.md:959]:
```suggestion
  - Internal Components
```

File: system-variables.md [hunk 2]
```diff
@@ -1160,6 +1160,10 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 >
 > Currently, PITR recovery handles the indexes created by index acceleration during the log backup with extra processing to achieve compatibility. For details, see [Why is the acceleration of adding indexes feature incompatible with PITR?](/faq/backup-and-restore-faq.md#why-is-the-acceleration-of-adding-indexes-feature-incompatible-with-pitr).
 
+> **Note:**
+>
+> Before you upgrade to v6.5.0 or later, it is recommended that you check whether the [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) path of TiDB is correctly mounted to an SSD disk. This parameter is a TiDB configuration item, which takes effect after TiDB is restarted. Therefore, setting this configuration item in advance before upgrading can avoid another restart.
```
Comment (qiancai) [inline system-variables.md:1165]:
```suggestion
> Before you upgrade TiDB to v6.5.0 or later, it is recommended that you check whether the [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) path of TiDB is correctly mounted to an SSD disk. This path is a TiDB configuration item, which takes effect after TiDB is restarted. Therefore, setting this configuration item before upgrading can avoid another restart.
```

File: system-variables.md [hunk 3]
```diff
@@ -1181,7 +1185,13 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Scope: GLOBAL
 - Persists to cluster: Yes
 - Default value: `OFF`
-- This variable is used to control whether to enable the distributed execution framework. After the framework is enabled, backend tasks such as DDL and import will be jointly completed by multiple TiDB nodes in the cluster.
+
+<CustomContent platform="tidb">
+
+- This variable is used to control whether to enable the [TiDB backend task distributed execution framework](/tidb-distributed-execution-framework.md). After the framework is enabled, backend tasks such as DDL and import will be jointly completed by multiple TiDB nodes in the cluster.
```
Comment (qiancai) [inline system-variables.md:1191]:
```suggestion
- This variable is used to control whether to enable the [TiDB backend task distributed execution framework](/tidb-distributed-execution-framework.md). After the framework is enabled, backend tasks such as DDL and import will be distributedly completed by multiple TiDB nodes in the cluster.
```

File: tidb-distributed-execution-framework.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+summary: Learn the use cases, limitations, usage, and implementation principles of the TiDB backend task distributed execution framework.
+---
+
+# TiDB Backend Task Distributed Execution Framework
+
+> **Warning:**
+>
+> This feature is an experimental feature. It is not recommended to use it in production environments.
+
+TiDB adopts a computing-storage separation architecture with excellent scalability and elasticity. Starting from v7.1.0, TiDB introduces a backend task distributed execution framework to further leverage the resource advantages of the distributed architecture. The goal of this framework is to implement unified scheduling and distributed execution of all backend tasks, and to provide unified resource management capabilities for the backend tasks, providing resource management capabilities for both the overall and individual backend tasks to better meet users' expectations for resource usage.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:12]:
```suggestion
TiDB adopts a computing-storage separation architecture with excellent scalability and elasticity. Starting from v7.1.0, TiDB introduces a backend task distributed execution framework to further leverage the resource advantages of the distributed architecture. The goal of this framework is to implement unified scheduling and distributed execution of all backend tasks, and to provide unified resource management capabilities for both overall and individual backend tasks, which better meets users' expectations for resource usage.
```

File: tidb-distributed-execution-framework.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+In a database management system, in addition to the core transactional processing (TP) and analytical processing (AP) workloads, there are other important tasks, such as DDL operations, Load Data, TTL, Analyze, and Backup/Restore, which are called **backend tasks**. These backend tasks need to process a large amount of data in database objects (tables), so they typically have the following characteristics:
+
+- Need to process all data in a schema or a database object (table).
+- Might need to be executed periodically, but at a low frequency.
+- If the resources are not properly controlled, they are prone to affect TP and AP tasks, lowering the database service quality.
+
+Enabling the TiDB backend task distributed execution framework can solve the above problems and has the following three advantages:
+
+- The framework provides unified capabilities for high scalability, high availability, and high performance.
+- The framework supports distributed execution of backend tasks, which can flexibly schedule the available computing resources of the entire TiDB cluster, thereby better utilizing the computing resources in the TiDB cluster.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:31]:
```suggestion
- The framework supports distributed execution of backend tasks, which can flexibly schedule the available computing resources of the entire TiDB cluster, thereby better utilizing the computing resources in a TiDB cluster.
```

File: tidb-distributed-execution-framework.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+- The framework provides unified resource usage and management capabilities, providing resource management capabilities for both the overall and individual backend tasks.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:32]:
```suggestion
- The framework provides unified resource usage and management capabilities for both overall and individual backend tasks.
```

Comment (qiancai) [reply tidb-distributed-execution-framework.md:32]: "provides unified resource usage and management capabilities" and "providing resource management capabilities" look a little duplicated.

File: tidb-distributed-execution-framework.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+Currently, the TiDB backend task distributed execution framework only supports the distributed execution of `ADD INDEX` statements, that is, the DDL statements that create indexes. For example, the following SQL statements are supported:
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:34]:
```suggestion
Currently, the TiDB backend task distributed execution framework only supports the distributed execution of `ADD INDEX` statements, that is, the DDL statements for creating indexes. For example, the following SQL statements are supported:
```

Comment (qiancai) [reply tidb-distributed-execution-framework.md:34]: 减少一个 that

File: tidb-distributed-execution-framework.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+CREATE INDEX idx1 ON table t1(c1);
+```
+
+## Prerequisites
+
+Before using the distributed framework, you need to enable the [Fast Online DDL](/system-variables.md#tidb_ddl_enable_fast_reorg-new-in-v630) mode.
+
+1. Adjust the following system variables related to Fast Online DDL:
+
+    * [`tidb_ddl_enable_fast_reorg`](/system-variables.md#tidb_ddl_enable_fast_reorg-new-in-v630): enabled by default starting from TiDB v6.5.0, used to enable Fast Online DDL mode.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:47]:
```suggestion
    * [`tidb_ddl_enable_fast_reorg`](/system-variables.md#tidb_ddl_enable_fast_reorg-new-in-v630): used to enable Fast Online DDL mode. It is enabled by default starting from TiDB v6.5.0.
```

File: tidb-distributed-execution-framework.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    * [`tidb_ddl_disk_quota`](/system-variables.md#tidb_ddl_disk_quota-new-in-v630): used to control the maximum quota of local disks that can be used in Fast Online DDL mode.
+
+2. Adjust the following configuration item related to Fast Online DDL:
+
+    * [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630): specifies the local disk path that can be used in Fast Online DDL mode.
+
+> **Note:**
+>
+> Before you upgrade to v6.5.0 or later, it is recommended that you check whether the [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) path of TiDB is correctly mounted to an SSD disk. This parameter is a TiDB configuration item, which takes effect after TiDB is restarted. Therefore, setting this configuration item in advance before upgrading can avoid another restart.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:56]:
```suggestion
> Before you upgrade TiDB to v6.5.0 or later, it is recommended that you check whether the [`temp-dir`](/tidb-configuration-file.md#temp-dir-new-in-v630) path of TiDB is correctly mounted to an SSD disk. This path is a TiDB configuration item, which takes effect after TiDB is restarted. Therefore, setting this configuration item in advance before upgrading can avoid another restart.
```

File: tidb-distributed-execution-framework.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+The architecture of the TiDB backend task distributed execution framework is as follows:
+
+![Architecture of the TiDB backend task distributed execution framework](/media/dist-task/dist-task-architect.jpg)
+
+As shown in the preceding diagram, the execution of backend tasks in the distributed framework is mainly handled by the following modules:
+
+- Dispatcher: generates the distributed execution plan for each task, manages the execution process, converts the task status, and collects and feeds back the runtime task information.
+- Scheduler: replicates the execution of distributed tasks among TiDB nodes to improve the efficiency of backend task execution.
+- Subtask Executor: the actual executor of distributed subtasks. In addition, the Subtask Executor returns the execution status of subtasks to the Scheduler, and the Scheduler updates the execution status of subtasks in a unified manner.
+- Resource pool: provides the basis for quantifying resource usage and management by pooling computing resources in the above modules.
```
Comment (qiancai) [inline tidb-distributed-execution-framework.md:90]:
```suggestion
- Resource pool: provides the basis for quantifying resource usage and management by pooling computing resources of the above modules.
```

**Review by qiancai — APPROVED** [review]
LGTM

File: tidb-distributed-execution-framework.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+As shown in the preceding diagram, the execution of backend tasks in the distributed framework is mainly handled by the following modules:
+
+- Dispatcher: generates the distributed execution plan for each task, manages the execution process, converts the task status, and collects and feeds back the runtime task information.
+- Scheduler: replicates the execution of distributed tasks among TiDB nodes to improve the efficiency of backend task execution.
+- Subtask Executor: the actual executor of distributed subtasks. In addition, the Subtask Executor returns the execution status of subtasks to the Scheduler, and the Scheduler updates the execution status of subtasks in a unified manner.
+- Resource pool: provides the basis for quantifying resource usage and management by pooling computing resources of the above modules.
+
+## See also
+
+* [Execution Principles and Best Practices of DDL Statements](/ddl-introduction.md)
```
Comment (ran-huang) [inline tidb-distributed-execution-framework.md:115]:
```suggestion
<CustomContent platform="tidb">

* [Execution Principles and Best Practices of DDL Statements](/ddl-introduction.md)

</CustomContent>
<CustomContent platform="tidb-cloud">

* [Execution Principles and Best Practices of DDL Statements](https://docs.pingcap.com/tidb/stable/ddl-introduction)

</CustomContent>
```

File: system-variables.md [hunk 12]
```diff
@@ -1181,7 +1185,7 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - Scope: GLOBAL
 - Persists to cluster: Yes
 - Default value: `OFF`
-- This variable is used to control whether to enable the distributed execution framework. After the framework is enabled, backend tasks such as DDL and import will be jointly completed by multiple TiDB nodes in the cluster.
+- This variable is used to control whether to enable the [TiDB backend task distributed execution framework](/tidb-distributed-execution-framework.md). After the framework is enabled, backend tasks such as DDL and import will be distributedly completed by multiple TiDB nodes in the cluster.
```
Comment (Benjamin2037) [inline system-variables.md:1188]:
```suggestion
- This variable is used to control whether to enable the [TiDB backend task distributed execution framework](/tidb-distributed-execution-framework.md). After the framework is enabled, backend tasks such as DDL and import will be distributedly executed and completed by multiple TiDB nodes in the cluster.
```

File: tidb-distributed-execution-framework.md [hunk 13]
```diff
@@ -0,0 +1,124 @@
...
+
+> **Warning:**
+>
+> This feature is an experimental feature. It is not recommended to use it in production environments.
+
+<CustomContent platform="tidb-cloud">
+
+> **Note:**
+>
+> Currently, this feature is only applicable to Dedicated Tier clusters. You cannot use it on Serverless Tier clusters.
```
Comment (Benjamin2037) [inline tidb-distributed-execution-framework.md:16]:
```suggestion
> Currently, this feature is only applicable to TiDB cloud Dedicated and on primise clusters . You cannot use it on TiDB cloud Serverless clusters.
```

**Review by Benjamin2037 — APPROVED** [review]
LGTM

File: tidb-distributed-execution-framework.md (hunk 13 above)
Comment (ran-huang) [reply tidb-distributed-execution-framework.md:16]: The content wrapped by `<CustomContent platform="tidb-cloud">` will only be seen by TiDB Cloud docs readers, so it's ok not to mention on-premises.

**Pull request merged**